repo/packages/J/jenkins/0/marathon.json

```diff
@@ -1,7 +1,7 @@
 {
   "id": "jenkins",
-  "instances": 1,
-  "mem": 1500,
+  "instances": 1.5,
+  "mem": 2048,
   "env": {
     "JAVA_OPTS": "-Xmx300m"
```

Comment:
cgroups have no impact on expansion of the JVM heap space.

If a VM is running out of heap space and can still grow according to it's MaxHeapSize it will try and expand, if this is above the limit of the cgroup the kernel will kill the process.

An `-Xms` and `-Xmx` should always be specified for java based apps running on mesos.  It is a good idea to leave some overhead between `-Xmx` and what is specified for `mem` to account for off heap space requirements (< 1.7 PermGen space).

To ensure that the heap arguments are applied to the java process the `cmd` value needs to change to:
`cd jenkins-on-mesos && export JENKINS_HOME=$(pwd) && export JENKINS_URL=http://$MESOS_HOSTNAME:$PORT0 && java $JAVA_OPTS -jar jenkins.war --httpPort=$PORT0`